eodatasets3/scripts/tostac.py

```diff
@@ -7,70 +7,182 @@
 from pathlib import Path
 from typing import Iterable
 from uuid import UUID
+from urllib.parse import urljoin
 
 import click
 import pyproj
-from click import echo
 from shapely.geometry.base import BaseGeometry
 from shapely.ops import transform
 
 from eodatasets3 import serialise
 from eodatasets3.ui import PathPath
 
 
+# Mapping between EO3 field names and STAC properties object field names
+MAPPING_EO3_TO_STAC = {
+    "dtr:end_datetime": "end_datetime",
+    "dtr:start_datetime": "start_datetime",
+    "eo:gsd": "gsd",
+    "eo:instrument": "instruments",
+    "eo:platform": "platform",
+    "eo:constellation": "constellation",
+    "eo:off_nadir": "view:off_nadir",
+    "eo:azimuth": "view:azimuth",
+    "eo:sun_azimuth": "view:sun_azimuth",
+    "eo:sun_elevation": "view:sun_elevation",
+}
+
+
+def convert_value_to_stac_type(key, value):
+    """
+    Convert return type as per STAC specification
+    :param key: Name of the field
+    :param value: Value of the field
+    :return: Converted value with required type in STAC
+
+    """
+    # In STAC spec, "instruments" have [String] type
+    if key == "eo:instrument":
+        return [value]
+    else:
+        return value
+
+
 @click.command(help=__doc__)
+@click.option("--stac-template", "-t", type=str)
+@click.option("--stac-base-url", "-u", default="", type=str)
+@click.option(
+    "--product-base-url", "-p", default="https://explorer.dea.ga.gov.au", type=str
```

Comment:
The code below assumes that this URL is an Explorer address (it's computing URL offsets from it that are specific to Explorer), so we probably should name the parameter more specifically: "explorer-url" or something

Reply:
Changed to `explorer-base-url`